contracts/src/octane/Preinstalls.sol

```diff
@@ -0,0 +1,128 @@
+    address internal constant SafeSingletonFactory = 0x914d7Fec6aaC8cd542e72Bca78B30650d45643d7;
+
+    /// @notice Address of the DeterministicDeploymentProxy predeploy.
+    address internal constant DeterministicDeploymentProxy = 0x4e59b44847b379578588920cA78FbF26c0B4956C;
+
+    /// @notice Address of the MultiSend_v130 predeploy.
+    address internal constant MultiSend_v130 = 0x998739BFdAAdde7C933B942a68053933098f9EDa;
+
+    /// @notice Address of the Permit2 predeploy.
+    address internal constant Permit2 = 0x000000000022D473030F116dDEE9F6B43aC78BA3;
+
+    /// @notice Address of the SenderCreator_v060 predeploy.
+    address internal constant SenderCreator_v060 = 0x7fc98430eAEdbb6070B35B39D798725049088348;
+
+    /// @notice Address of the EntryPoint_v060 predeploy.
+    address internal constant EntryPoint_v060 = 0x5FF137D4b0FDCD49DcA30c7CF57E578a026d2789;
+
+    /// @notice Address of the SenderCreator_v070 predeploy.
+    address internal constant SenderCreator_v070 = 0xEFC2c1444eBCC4Db75e7613d20C6a62fF67A167C;
+
+    /// @notice Address of the EntryPoint_v070 predeploy.
+    address internal constant EntryPoint_v070 = 0x0000000071727De22E5E9d8BAf0edAc6f37da032;
+
+    /// @notice Address of beacon block roots contract, introduced in the Cancun upgrade.
+    ///         See BEACON_ROOTS_ADDRESS in EIP-4788.
+    ///         This contract is introduced in L2 through an Ecotone upgrade transaction, if not already in genesis.
+    /// @dev Note that on Omni EVM, the BeaconBlockRoots.get() returns the AppHash of the Omni's
+    ///      consensus chain. Omni does not have a beacon chain.
+    address internal constant BeaconBlockRoots = 0x000F3df6D732807Ef1319fB7B8bB8522d0Beac02;
+
+    /// @notice See https://eips.ethereum.org/EIPS/eip-4788, this is the address of the sender of the deployment tx.
+    /// The nonce of this account must be non-zero, to ensure the Ecotone upgrade tx is still successful
+    /// if the code is already in place.
+    address internal constant BeaconBlockRootsSender = 0x0B799C86a49DEeb90402691F1041aa3AF2d3C875;
+
+    /// @notice Address of the ERC1820Registry predeploy.
+    address internal constant ERC1820Registry = 0x1820a4B7618BdE71Dce8cdc73aAB6C95905faD24;
+
+    // @notice Permit2 code is templated. The template is a copy of the Mainnet Ethereum L1 Permit2 deployment.
+    //         This deployed bytecode contains two immutable values _CACHED_CHAIN_ID and _CACHED_DOMAIN_SEPARATOR,
+    //         which depend on the chain-ID (the domain-separator is computed with the chain-ID as input).
+    //         Note that the _CACHED_DOMAIN_SEPARATOR in the Permit2 deployed-bytecode of the
+    //         Permit2 test-utils is wrong, and does not match the chain-ID of 31337 in said bytecode.
+    //         These two values are substituted for accuracy when the code is retrieved with getPermit2Code(chainid).
+    bytes internal constant Permit2TemplateCode =
```

Comment:
can we get some auditability into how these are generated?

Reply:
will audit / document with tests